### DISCUSSION TITLE
Run monailabel on my own dataset

### DISCUSSION THREAD

**Discussion by Yangtting** [Project-MONAI/MONAILabel#1607] [Q&A] [answered]
Hello! First of all, thank you for developing such a useful plugin as Monailabel! I'm really interested about it, and I configured my environment on Windows followed the tutorial and successfully downloaded monailabel as well as 3D slicer. 

Then I created my own dataset just like Task09_Spleen dataset. Due to that I have already trained my own dataset, I put the pretrained model named 'pretrained_segmentation.pt' trained by UNet into the directory 'C:\Users\TRY\apps\radiology\model', and also change the segmentation.py in ‘C:\Users\TRY\apps\radiology\lib\configs’ to fit my task.

However, When I clicked the auto-segmentation button, the results is very strange unlike the results by pretrained model. Could you please tell me how to solve this problem? Thanks very much!

Code:
`monailabel start_server --app C:/Users/TRY/apps/radiology --studies C:/Users/TRY/datasets/Task_RH/imagesTr --conf models segmentation`

Change segmenttaion.py in configs directory
![微信图片_20231211190101](https://github.com/Project-MONAI/MONAILabel/assets/75001245/2481bd48-50a7-4376-9f38-ff7165b9e114)

![微信图片_20231211191043](https://github.com/Project-MONAI/MONAILabel/assets/75001245/986a891b-0994-4df0-b604-46b37924a5d2)

**Comment by Yangtting** [comment]
I just found that the monailabel didn't include the ScaleIntensityRanged function which I used in the pretraining process, I guess it is the reason that the output in 3D Slicer is really strange! Could you please tell me how to add this fuction in the segmentation.py code! Thanks really much!

**Reply by Yangtting** [reply]
Sorry for borthering you! I had sovled it! I added this fuction in segmentation.py under infers directory.

**Reply by diazandr3s** [reply, marked as answer]
Hi @Yangtting,

Thanks for posting this question and updating it as well with your experience.

> I added this fuction in segmentation.py under infers directory.

That's correct, the pre-processing transforms for inference can be found under the [lib/infers](https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/radiology/lib/infers) folder. The same applies to the inferrer arguments: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/segmentation.py#L77
